src/wallet/sqlite.cpp

```diff
@@ -15,18 +16,66 @@
 
 static const char* DATABASE_FILENAME = "wallet.dat";
 
+static Mutex g_sqlite_mutex;
+static int g_sqlite_count GUARDED_BY(g_sqlite_mutex) = 0;
+
+static void ErrorLogCallback(void* arg, int code, const char* msg)
+{
+    // From sqlite3_config() documentation for the SQLITE_CONFIG_LOG option:
+    // "The void pointer that is the second argument to SQLITE_CONFIG_LOG is passed through as
+    // the first parameter to the application-defined logger function whenever that function is
+    // invoked."
+    // Assert that this is the case:
+    assert(arg == nullptr);
```

Comment:
70a2c658549c8a0caa3e58dcc7894d2bb8872cd5
nit: Maybe use (here and in other commits) the new `Assert` (#19277,  #19491)?

Reply:
That does the opposite of what we want.